api/src/main/java/org/openmrs/CohortMembership.java

```diff
@@ -144,4 +144,18 @@ public int compareTo(CohortMembership o) {
 		}
 		return ret;
 	}
+	
+	/**
+	 * @since 2.3.0
+	 * Indicates if a given cohortMembership object is equal to this one
+	 * 
+	 * @param otherCohortMembership is a CohortMembership object that should be checked for equality with this object
+	 * @return true if both objects are logically equal. This is the case when endDate, startDate and patientId are equal  
+	 */
+	public boolean equals(CohortMembership otherCohortMembership) {
+		if(this == otherCohortMembership) return true;
+		
+		return otherCohortMembership != null && this.endDate.equals(otherCohortMembership.getEndDate()) && this.startDate.equals(otherCohortMembership.getStartDate()) 
```

Comment:
What happens if any of these properties is null?

Reply:
You are right! I did not took this into account. I changed the code and added two new tests. Are two tests alright or should I add further test cases dealing with null values?

Reply:
Those tests would not hurt. 😊

Reply:
All possible cases should be tested now!